src/components/Carousel.jsx

```diff
@@ -0,0 +1,126 @@
+import React, { useState, useEffect } from "react";
+import {
+  Paper,
+  Typography,
+  Grid,
+  IconButton,
+  Card,
+  CardMedia,
+  CardContent,
+  Box,
+} from "@mui/material";
+import KeyboardArrowLeftIcon from "@mui/icons-material/KeyboardArrowLeft";
+import KeyboardArrowRightIcon from "@mui/icons-material/KeyboardArrowRight";
+import ModalDetails from "./ModalDetailes";
+
+const cardTextStyle = {
+  backgroundColor: "rgba(24, 20, 20, 0.7)",
+  color: "#fff",
+  padding: "8px",
+  position: "absolute",
+  bottom: "0",
+  left: "0",
+  right: "0",
+};
+
+function Carousel({ getFunction, title, genreId, apiFetched }) {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [selectedItem, setSelectedItem] = useState(null);
+  const [itemsData, setItemsData] = useState([]);
+
+  const itemsPerPage = 8;
+  const dynamicNumberOfItems = 32;
+
+  useEffect(() => {
+    if (getFunction) {
+      const items =
+        genreId !== null
+          ? getFunction(dynamicNumberOfItems, genreId)
+          : getFunction(dynamicNumberOfItems);
+      const parsedItems = items.map((jsonString) => JSON.parse(jsonString));
+      setItemsData(parsedItems);
+    } else if (apiFetched) {
+      setItemsData(apiFetched);
+    }
+  }, [getFunction, apiFetched, genreId]);
+  const totalObjects = itemsData.length;
+  const totalPages = Math.ceil(totalObjects / itemsPerPage);
+
+  const startIndex = (currentPage - 1) * itemsPerPage;
+  const endIndex = Math.min(startIndex + itemsPerPage, totalObjects);
+
+  const handlePrevClick = () => {
```

Comment:
well, nicely done
try and achive it in one liner:
```js
const handlePrevClick = () => setCurrentPage((current) => current > 1 ? current - 1 : totalPages)
```

Reply:
hmmm, i actully decided to handlePageChange by in other way.
instead of 2 handlePrevClick  and handleafterClick  (dont remember the name) 
there is only handlePageChange, and it looks good...i think